### PR TITLE
chore: Change CLI and constants

### PR DESCRIPTION
**Pull request by szewaiyuen6**
## Description

Change user-facing parts of the det deploy that references lore to genai.

Internal references clean up will be a separate task (lower pri).

## Test Plan

`det deploy aws up --cluster-id swy-lore-test4   --region us-west-2 --keypair szewai-determined-aws --compute-agent-instance-type g4dn.12xlarge --max-dynamic-agents 10 --det-v 115f685a4cd4025118a198751a81b5e0105c5f68  --max-idle-agent-period 60m   --deployment-type genai --genai-version a45d508`

Screen Shot 2023-12-13 at 12 57 29 PM

Screen Shot 2023-12-13 at 12 58 04 PM

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by hamidzr — COMMENTED** [review]
thanks for working on this. 

leaving the internal references to lore as lore for as a separate tasks sgtm but I think we should at least migration all the `lore.?version` references together in this pr otherwise it'd be hard to track

```
(determined) hbp ➜ determined git:(update_deploy_2) ag 'lore.?version'
harness/determined/deploy/aws/deployment_types/vpc.py
49:        lore_tag = self.parameters[constants.cloudformation.LORE_VERSION] or "latest"

harness/determined/deploy/aws/constants.py
78:    LORE_VERSION = "LoreVersion"

harness/determined/deploy/aws/cli.py
270:        constants.cloudformation.LORE_VERSION: args.genai_version,

harness/determined/deploy/aws/templates/lore.yaml
219:  LoreVersion:
221:    Description: Lore version to deploy
294:      {{ if ne .lore_version "" }}
922:            lore_version: ${LoreVersion}
974:            if [ ! -z "${LoreVersion}" ]; then
983:                --env LORE_DOCKER_TAG_SUFFIX="${LoreVersion}" \
987:                "determinedai/environments-dev:lore-backend-image-${LoreVersion}"

harness/determined/deploy/aws/deployment_types/base.py
45:    constants.cloudformation.LORE_VERSION,

harness/determined/deploy/aws/templates/fsx.yaml
289:      {{ if ne .lore_version "" }}

harness/determined/deploy/aws/templates/simple.yaml
280:      {{ if ne .lore_version "" }}

harness/determined/deploy/aws/templates/efs.yaml
289:      {{ if ne .lore_version "" }}

harness/determined/deploy/aws/templates/govcloud.yaml
234:      {{ if ne .lore_version "" }}

harness/determined/deploy/aws/templates/simple-rds.yaml
288:      {{ if ne .lore_version "" }}

harness/determined/deploy/aws/templates/secure.yaml
284:      {{ if ne .lore_version "" }}

harness/determined/deploy/aws/injector/master.yaml.tmpl
60:{{ if ne .lore_version "" }}
(determined) hbp ➜ determined git:(update_deploy_2)
```

**Review by hamidzr — APPROVED** [review]
lgtm assuming it deploys fine.

we probably want to consider this env var user facing but it needs to be updated on the other side 
`harness/determined/deploy/aws/templates/lore.yaml:983:                --env LORE_DOCKER_TAG_SUFFIX="${GenAIVersion}" \
`

File: harness/determined/deploy/aws/deployment_types/vpc.py [hunk 1]
```diff
@@ -42,10 +42,12 @@ class EFS(VPCBase):
 
 class Lore(VPCBase):
```
Comment (azhou-determined) [inline harness/determined/deploy/aws/deployment_types/vpc.py:43]: should this classname be changed too?

**Review by azhou-determined — APPROVED** [review]
lgtm

File: harness/determined/deploy/aws/deployment_types/vpc.py (hunk 1 above)
Comment (szewaiyuen6) [reply harness/determined/deploy/aws/deployment_types/vpc.py:43]: This is an internal reference, we are doing the naming migration in phases and focusing on user facing noes first.

**Pull request merged**